Sources/FoundationEssentials/UUIDv4.swift

```diff
@@ -0,0 +1,52 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift open source project
+//
+// Copyright (c) 2022 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+//
+//===----------------------------------------------------------------------===//
+
+extension UUID {
+    static func v4_generatedRandom() -> uuid_t {
+        var randomBits = (0 ... 15).map { _ in UInt8.random(in: .min ... .max) }
+        randomBits[6] = (randomBits[6] & 0x0F) | 0x40
+        randomBits[8] = (randomBits[8] & 0x3F) | 0x80
+
+        return randomBits.withUnsafeBytes { buffer in
+            return buffer.bindMemory(to: uuid_t.self)[0]
+        }
+    }
+
+    static func v4_parse(uuidString string: __shared String) -> uuid_t? {
+        let components = string
```

Comment:
Based on https://github.com/apple/swift-foundation/pull/129#issuecomment-1577124404, one obvious optimization we can do here is to `guard` against `string.count`. However, both `String.count` and `String.UTF8View.count` is not O(1), so I guess we should parse the string manually and `throw` if it’s malformed, including too short or too long?